app/src/main/java/com/sun/efood/data/source/remote/MealRemoteDataSource.kt

```diff
@@ -0,0 +1,59 @@
+package com.sun.efood.data.source.remote
+
+import com.sun.efood.data.source.MealDataSource
+import com.sun.efood.data.source.remote.api.ApiConstants
+import com.sun.efood.data.source.remote.utils.connectData
+import com.sun.efood.data.source.utils.LoadDataAsync
+import com.sun.efood.data.source.utils.OnDataCallback
+import com.sun.efood.data.model.Meal
+import com.sun.efood.data.model.MealDetail
+import com.sun.efood.data.source.remote.api.ApiQuery
+import com.sun.efood.utils.Constants
+import com.sun.efood.utils.parseJsonToObject
+import org.json.JSONObject
+
+@Suppress("DEPRECATION")
+class MealRemoteDataSource : MealDataSource.Remote {
+
+    override fun getMealByCategory(nameCategory: String, callback: OnDataCallback<List<Meal>>) {
+        LoadDataAsync<Unit, List<Meal>>(callback) {
+            getMealByCategory(nameCategory)
+        }.execute(Unit)
+    }
+
+    override fun searchMeal(nameMeal: String, callback: OnDataCallback<List<Meal>>) {
+        LoadDataAsync<Unit, List<Meal>>(callback) {
+            getMealByKeySearch(nameMeal)
+        }.execute(Unit)
+    }
+
+    override fun getMealDetailByMeal(nameMeal: String, callback: OnDataCallback<List<MealDetail>>) {
+        LoadDataAsync<Unit, List<MealDetail>>(callback) {
+            getMealDetailByMeal(nameMeal)
+        }.execute(Unit)
+    }
+
+    private fun getMealByCategory(nameCategory: String): List<Meal> =
+        JSONObject(connectData(ApiQuery.queryMealByCategory(nameCategory)))
+            .getString(Constants.MEALS).parseJsonToObject()
+
+    private fun getMealDetailByMeal(idMeal: String): List<MealDetail> =
+        JSONObject(connectData(ApiQuery.queryMealDetail(idMeal)))
+            .getString(Constants.MEALS).parseJsonToObject()
```

Comment:
getString khác optString chỗ nào đại ca chỉ em

Reply:
getString dùng với kiểu NonNull còn optString dùng với kiểu Nullable
https://stackoverflow.com/questions/13790726/the-difference-between-getstring-and-optstring-in-json/13790789

Reply:
Giải thích thế nói làm gì :)) đang kiểm tra kiến thức chứ a có hỏi bài m đâu :v

Reply:
Thì em có biết đâu, e phải search gg đấy chứ :v